### PR TITLE
separate out and improve the config reading

### PR DESCRIPTION
**Pull request by freben**
Broke out the expansion of shorthands into a function of its own, cleaned up the code a bit, made more expressive error messages, and wrote more tests

### CODE REVIEW

File: packages/frontend-app-api/src/wiring/parameters.ts [hunk 1]
```diff
@@ -40,93 +49,210 @@ export function readAppExtensionParameters(
...
-      config: value.config /* validate later */,
+      id: key,
+      extension: resolveExtensionRef(value),
     };
-  });
+  }
+
+  // The remaining case is the generic object. Example YAML:
+  // - core.router/routes:
+  //     id: redirects.catalog
```
Comment (Rugvip) [inline packages/frontend-app-api/src/wiring/parameters.ts:155]: Let's not allow an ID here imo, if one wants to specify an ID they should use it as a key. This effectively just ends up being an alternate format where id and at are flipped

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice :+1: 

Just thinking ID shouldn't be an allowed key

File: packages/frontend-app-api/src/wiring/parameters.ts (hunk 1 above)
Comment (freben) [reply packages/frontend-app-api/src/wiring/parameters.ts]: Aye, a worthwhile simplification. Fixed

File: packages/frontend-app-api/src/wiring/parameters.ts [hunk 2]
```diff
@@ -19,6 +19,16 @@ import {
   Extension,
   ExtensionInstanceParameters,
 } from '@backstage/frontend-plugin-api';
+import { JsonValue } from '@backstage/types';
+import omitBy from 'lodash/omitBy';
+
+const knownExtensionInstanceParameters = [
+  'id',
```
Comment (Rugvip) [inline packages/frontend-app-api/src/wiring/parameters.ts:26]: Nope!

Comment (freben) [reply packages/frontend-app-api/src/wiring/parameters.ts]: aw shucks! good catch, removed and updated test

**Review by Rugvip — APPROVED** [review]

**Pull request merged**